avocado/core/suite.py

```diff
@@ -0,0 +1,141 @@
+
+
+class TestSuiteError(Exception):
+    pass
+
+
+class TestSuiteStatus(Enum):
+    RESOLUTION_NOT_STARTED = 0
+    TESTS_NOT_FOUND = 1
+    TESTS_FOUND = 2
+    UNKNOWN = 3
+
+
+class TestSuite:
+    def __init__(self, name, config, tests=None):
+        self.name = name
+        self.tests = tests or None  # this will force None when tests = []
+
+        # Create a complete config dict with all registered options + custom
+        # config
+        self.config = settings.as_dict()
+        if config:
+            self.config.update(config)
+
+        self._variant = None
+        self._references = None
+
+        if config.get('run.dry_run.enabled'):
+            self.convert_to_dry_run()
+
+    def __len__(self):
+        """This is a convenient method to run `len()` over this object.
+
+        With this you can run: len(a_suite) and will return the same as
+        `len(a_suite.tests)`.
+        """
+        return self.size
+
+    def _parse_variant(self):
+        # Varianter not yet parsed, apply configs
+        if not self.variant.is_parsed():
+            try:
+                self.variant.parse(self.config)
+            except (IOError, ValueError) as details:
+                raise OptionValidationError("Unable to parse "
+                                            "variant: %s" % details)
+
+    @property
+    def references(self):
+        if self._references is None:
+            self._references = self.config.get('run.references')
+        return self._references
+
+    @property
+    def size(self):
+        """The overall length/size of this test suite."""
+        if self.tests is None:
+            return 0
+        return len(self.tests)
+
+    @property
+    def status(self):
+        if self.tests is None:
+            return TestSuiteStatus.RESOLUTION_NOT_STARTED
+        elif self.size == 0:
+            return TestSuiteStatus.TESTS_NOT_FOUND
+        elif self.size > 0:
+            return TestSuiteStatus.TESTS_FOUND
+        else:
+            return TestSuiteStatus.UNKNOWN
+
+    @property
+    def variant(self):
+        if self._variant is None:
+            self._variant = self.config.get("avocado_variants") or Varianter()
+        self._parse_variant()
+        return self._variant
+
+    def convert_to_dry_run(self):
+        for i in range(self.size):
+            self.tests[i] = [DryRunTest, self.tests[i][1]]
+
+    @classmethod
+    def from_config(cls, config, name=None, runner='nrunner'):
+        references = config.get('run.references')
+        if not references:
+            msg = ("Test Suite could not be create. No test references "
+                   "provided nor any other arguments resolved into tests")
+            raise TestSuiteError(msg)
+
+        if runner == 'nrunner':
+            return cls.from_config_with_resolver(config, name)
+        else:
+            return cls.from_config_with_loader(config, name)
+
+    @classmethod
+    def from_config_with_loader(cls, config, name=None):
```

Comment:
*Maybe*, in accordance to the comment on the `convert_to_dry_run()` not being a public interface, this (and `from_config_with_resolver()`) should also become private.  Just an idea as we evolve the interface and think about the usability aspects.